platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/FileOrUriNotationConverter.java

```diff
@@ -122,6 +127,17 @@ public void convert(Object notation, NotationConvertResult<? super Object> resul
         }
     }
 
+    private static boolean tryConvertToFile(URI uri, NotationConvertResult<? super Object> result) {
```

Comment:
IMO passing the `result` into this method makes the control flow difficult to follow. I had originally not seen that this method has side-effects and calls `converted`. I think this would be much easier to follow if this just returned a nullable file and the side-effects were handled outside of this method